src/Chapter2.hs

```diff
@@ -728,7 +747,8 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate = foldr (\x -> (++) (replicate x x)) []
```

Comment:
Interesting solution ⭐ 

Alternatively, you could use `concatMap` function here 🙂